### DISCUSSION TITLE
Issue with Shell Tests

### DISCUSSION THREAD

**Discussion by SeanAlling-DojoFive** [sierrafoxtrot/srecord#44] [Ideas]
I have run into issues when running `ctest` multiple times on same build will result in different test failing. 

In my case I can run `ctest` and if a failed test, I can run `ctest` again and get a different set of failed tests and failed tests from first run now pass. 

I believe the issue is related to the random generation of data. 

Has anyone else run into or seen the same issue?

**Comment by jtxa** [comment]
Never had such issue, and don't see any reason for that. GH Actions also never failed randomly (Ubuntu, MSYS on Windows and MacOS).
The shell scripts are there since the beginning of SRecord, and CTest does not much more than just calling them.

The testdir is `/tmp/srecord-$$`, where `$$` is the current PID of the shell (which is unique on the system). So even parallelized calls by CTest should not matter.

Do you have a list of numbers which fail sometimes for you? Anything special on your host system?
